### ISSUE TITLE
When an existing customer upgrades from NPSP 2 and has Record Type values by name in Custom Settings, they're no longer getting converted to IDs

### ISSUE THREAD

**Issue by judisohn** [SalesforceFoundation/Cumulus#2890]
There have been multiple reports of this in the Hub over the past few weeks. This is causing the inability to load NPSP Settings, errors in Health Check, etc.

Example [here](https://powerofus.force.com/0D58000003s0jTX) and [here](https://powerofus.force.com/0D58000003rjlhu) and [here](https://powerofus.force.com/0D58000003nNGD8) and [here](https://powerofus.force.com/0D58000003nLqrT). 

resolution:

Click Manage to the left of Contacts & Orgs Settings. Make sure 'Excluded Opp Record Types' 'Household Account Record Type' and 'One-to-One Record Type' do not have the word 'Membership.'

If that's good, go to Households Settings, click Manage to the left of that and check for the word 'Membership' in 'Household Creation Excluded Record Types' 'Membership Record Type Names' (this is where I think the problem may be) 'Opp RecTypes Excluded for Account Rollup' 'Opp RecTypes Excluded for Contact Rollup' or 'Household OCR Excluded Record Types'

Basically, NPSP now requires an ID in those fields rather than a name. I suspect that for whatever reason, you had the name and it's not mapping correctly to the record type so it didn't update to the correct name.

**Comment by judisohn** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x800000093KRSAA2" target="_blank">W-021618</a>

**Comment by cdcarter** [comment]
https://github.com/SalesforceFoundation/Cumulus/issues/2862 related

**Comment by judisohn** [comment]
To resolve this, go to Setup and search for Custom Settings. Click Manage to the right of the listed custom setting, then Edit, and delete any value in any of these listed fields. These are record types which you can go back to NPSP Settings afterwards and re-select (so make note of the names if names are listed). The error is because the ID is invalid or it's a text value that wasn't converted to a record type ID properly.

from Contacts & Orgs Settings:
- Household Account Record Type
- One-to-One Record Type
- Excluded Opp Record Types

from Household Settings:
- Membership Record Type Names
- Household Creation Excluded Record Types
- Household OCR Excluded Record Types
- Opp RecTypes Excluded for Account Rollup
- Opp RecTypes Excluded for Contact Rollup

**Comment by mrbelvedere** [comment]
Included in beta release 3.122 (Beta 4)

**Comment by mrbelvedere** [comment]
Included in production release rel/3.122
